test.py

```diff
@@ -2204,3 +2210,291 @@ def test_platform_apple():
+    assert unredeemed_payment.expiry_unix_ts_ms ==expiry_time_unix_ms
+    assert unredeemed_payment.grace_period_duration_ms == 0
+    assert unredeemed_payment.platform_refund_expiry_unix_ts_ms == platform_refund_expiry_unix_ms
+    assert unredeemed_payment.refunded_unix_ts_ms == None
+    assert unredeemed_payment.apple == backend.AppleTransaction()
+    assert unredeemed_payment.google_payment_token == purchase_token
+    assert unredeemed_payment.google_order_id == order_id
+    
+    version: int                            = 0
+    add_pro_payment_tx                      = backend.AddProPaymentUserTransaction()
+    add_pro_payment_tx.provider             = unredeemed_payment.payment_provider
+    add_pro_payment_tx.google_payment_token = purchase_token
+
+    payment_hash_to_sign: bytes = backend.make_add_pro_payment_hash(version=version,
+                                                                    master_pkey=master_key.verify_key,
+                                                                    rotating_pkey=rotating_key.verify_key,
+                                                                    payment_tx=add_pro_payment_tx)
+
+    request_body={
+          'version':              version,
+          'master_pkey':          bytes(master_key.verify_key).hex(),
+          'rotating_pkey':        bytes(rotating_key.verify_key).hex(),
+          'master_sig':           bytes(master_key.sign(payment_hash_to_sign).signature).hex(),
+          'rotating_sig':         bytes(rotating_key.sign(payment_hash_to_sign).signature).hex(),
+          'payment_tx': {
+              'provider':             add_pro_payment_tx.provider.value,
+              'google_payment_token': add_pro_payment_tx.google_payment_token,
+          }
+        }
+    response: werkzeug.test.TestResponse = flask_client.post(server.FLASK_ROUTE_ADD_PRO_PAYMENT, json=request_body)
+
+    unredeemed_payments = backend.get_unredeemed_payments_list(db.sql_conn)
+    assert len(unredeemed_payments) == 0
+
+    payments = backend.get_payments_list(sql_conn=db.sql_conn)
+    assert len(payments) == 1
+
+    payment = payments[0]
+    assert isinstance(payment, backend.PaymentRow)
+    assert payment.master_pkey == bytes(master_key.verify_key)
+    assert payment.status == backend.PaymentStatus.Redeemed
+    assert payment.plan == backend.ProPlanType.OneMonth
+    assert payment.payment_provider == base.PaymentProvider.GooglePlayStore
+    assert payment.redeemed_unix_ts_ms is not None and payment.redeemed_unix_ts_ms > event_ms and payment.redeemed_unix_ts_ms == base.round_unix_ts_ms_to_next_day(event_ms)
+    assert payment.expiry_unix_ts_ms == unredeemed_payment.expiry_unix_ts_ms
+    assert payment.grace_period_duration_ms == 0
+    assert payment.platform_refund_expiry_unix_ts_ms == platform_refund_expiry_unix_ms
+    assert payment.refunded_unix_ts_ms == None
+    assert unredeemed_payment.apple == backend.AppleTransaction()
```

Comment:
Is this meant to be assertions for `payment.apple` and `payment.google_payment_token`?